dmcontent/questions.py

```diff
@@ -41,10 +41,26 @@ def filter(self: TQuestion, context, dynamic=True, inplace_allowed: bool = False
 
     def _should_be_shown(self, context):
         return all(
-            depends["on"] in context and (context[depends["on"]] in depends["being"])
+            self._is_dependency_valid(depends["on"], depends["being"], context)
             for depends in self.get("depends", [])
         )
 
+    def _is_dependency_valid(self, key, condition, context):
```

Comment:
I know that content loader's performance strongly affects the apps (see all the work done to enable in-place filtering). Can we improve our performance of handling something like:

```
                "on": "other_number",
                "being": [
                    {"min_value": 0, "max_value": 0, "type": "integer"}
                    {"min_value": 0, "max_value": 0, "type": "integer"}
                    {"min_value": 0, "max_value": 0, "type": "integer"}
                    {"min_value": 0, "max_value": 0, "type": "integer"}
                    {"min_value": 0, "max_value": 0, "type": "integer"}
                    {"min_value": 0, "max_value": 0, "type": "integer"}
                    {"min_value": 0, "max_value": 0, "type": "integer"}
                    {"min_value": 0, "max_value": 0, "type": "integer"}
                    {"min_value": 0, "max_value": 0, "type": "integer"}
                    {"min_value": 0, "max_value": 0, "type": "integer"}
                    {"min_value": 0, "max_value": 0, "type": "integer"}
                    {"min_value": 0, "max_value": 0, "type": "integer"}
                    {"min_value": 0, "max_value": 0, "type": "integer"}
                    {"min_value": 0, "max_value": 0, "type": "integer"}
                    {"min_value": 0, "max_value": 0, "type": "integer"}
                ]
```